### PR TITLE
Improvement/endpoints

### LINKED ISSUE #145: change endpoints to not return children
make `/networks` and `/devices` not return all their children
maybe change current design to a different endpoint

### PR DESCRIPTION
**Pull request by Shiny380**
Resolves #145 
**ALL** `network` and `device` endpoints now accept `with_children` argument in the query string
i.e.
- `/api/modbus/network?with_children=true` will not return all children
- `/api/modbus/network` will not return just the network data

Depends on #152

### CODE REVIEW

File: src/resources/utils.py
Comment (RaiBnod) [inline src/resources/utils.py:16]: Use logger instead of print.

File: src/resources/resource_device.py
Comment (RaiBnod) [inline src/resources/resource_device.py:10]: We can use decorator for such cases, but fine for now.

**Review by RaiBnod — APPROVED** [review]
Nothing special changes requested, just suggestions. So you can merge this one after your previous pull request gets merged.

File: src/resources/resource_device.py
Comment (Shiny380) [reply src/resources/resource_device.py:10]: nah I didn't use the decorator as I return 2 different marshallings depending on the query string

**Pull request merged**
